<?xml version="1.0"?>
<launch> 
  
  <!-- MAKE SURE TO SWITCH TO INTENSITY BRANCH TO KEEP INTENSITIES IN ACCUMULATED CLOUD -->
  <!-- Accumulated pointcloud -->
 <node pkg="vigir_cloud_to_planar_scan" type="periodic_cloud_aggregator_node" name="lidar_aggregator_node" output="screen">
    <remap from="cloud" to="/spin_laser/vlp16"/>
    <remap from="cloud_out" to="/spin_laser/vlp16_agg"/> 
    <param name="target_frame" value="base_link"/>
    <param name="publish_frequency_hz" value="0.05"/>
  </node>   
  
  <!-- Capture data -->
  <node pkg="hector_camera_lidar_calibration" type="data_collector_node" name="camera_lidar_calibration_data_collector_node" output="screen">
    <rosparam file="$(find jasmine_robot_bringup)/config/ricoh_theta/cameras.yaml" command="load" />
    
    <param name="base_frame" value="base_link"/>
    <param name="cam_head_frame" value="ricoh_theta_center_link"/>
    <param name="bin_fraction" value="1"/>
    <param name="mask_path" value="$(find hector_camera_lidar_calibration)/masks"/>
    
    <remap from="cloud" to="/spin_laser/vlp16_agg"/>
  </node>
</launch>
